Doc/library/signal.rst

```diff
@@ -273,20 +273,22 @@ The :mod:`signal` module defines the following functions:
    .. versionadded:: 3.3
 
 
-.. function:: setitimer(which, seconds[, interval])
+.. function:: setitimer(which, seconds, interval=0.0)
 
    Sets given interval timer (one of :const:`signal.ITIMER_REAL`,
    :const:`signal.ITIMER_VIRTUAL` or :const:`signal.ITIMER_PROF`) specified
    by *which* to fire after *seconds* (float is accepted, different from
    :func:`alarm`) and after that every *interval* seconds. The interval
-   timer specified by *which* can be cleared by setting seconds to zero.
+   timer specified by *which* can be cleared by setting *seconds* to zero.
 
    When an interval timer fires, a signal is sent to the process.
    The signal sent is dependent on the timer being used;
    :const:`signal.ITIMER_REAL` will deliver :const:`SIGALRM`,
    :const:`signal.ITIMER_VIRTUAL` sends :const:`SIGVTALRM`,
    and :const:`signal.ITIMER_PROF` will deliver :const:`SIGPROF`.
 
+   If *interval* is equal to zero, the signal will only be sent once.
```

Comment:
I'm not sure that it's correct to promise that the signal *will* be sent. If *seconds* is in the past (using ITIMER_REAL), the signal is never sent, no?

Maybe I should replace "and after that every *interval* seconds" with "and after that every *interval* seconds (if *interval* is not equal to zero)" instead? What do you think @pitrou?

Reply:
According to [POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/functions/setitimer.html), `setitimer` should return EINVAL when a negative argument is passed.

Reply:
```python
>>> signal.setitimer(signal.ITIMER_REAL, 0, -1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
signal.ItimerError: [Errno 22] Invalid argument
>>> signal.setitimer(signal.ITIMER_REAL, -1, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
signal.ItimerError: [Errno 22] Invalid argument
```

Reply:
I'm not sure that I understand correctly your comments. Do you mean that setitimer() warrants that at least one signal will be fired?

I changed the documentation to not overspecify Python since Python is just a thin wrapper to the system setitimer() function. I just added "(if *interval* is non-zero)" to the existing doc.